File: 1431-kids-with-the-greatest-number-of-candies/main.go
func kidsWithCandies(candies []int, extraCandies int) []bool {
    
    var ans []bool
    max := max(candies)
    for _, kid := range candies {
        ans = append(ans, (kid + extraCandies >= max))
    }

    return ans
}


func max(nums []int) int {
    maxValue := 0
    for _, num := range nums {
        if num > maxValue {
            maxValue = num
        }
    }
    return maxValue
}

File: 1431-kids-with-the-greatest-number-of-candies/main.rs
impl Solution {
    pub fn kids_with_candies(candies: Vec<i32>, extra_candies: i32) -> Vec<bool> {
        
        let mut ans: Vec<bool> = Vec::with_capacity(candies.len());
        let max = match candies.iter().max() {
            Some(&max_value) => max_value,
            None => return ans,
        };

        for &candie in candies.iter() {
            ans.push(candie + extra_candies >= max);
        };

        ans
    }
}
